Source/Rhetos.Extensibility/PluginScanner.cs

```diff
@@ -163,18 +163,65 @@ private List<string> ListAssemblies()
             return pluginsByExport;
         }
 
-        private static Dictionary<Type, List<PluginInfo>> GetMefExportsForAssembly(string assemblyPath, List<string> typesToCheck = null)
+        private Dictionary<Type, List<PluginInfo>> GetMefExportsForAssembly(string assemblyPath, List<string> typesToCheck = null)
         {
             if (typesToCheck != null && typesToCheck.Count == 0) return new Dictionary<Type, List<PluginInfo>>();
 
-            var assembly = Assembly.LoadFrom(assemblyPath);
+            var assembly = LoadAssembly(assemblyPath);
             var types = typesToCheck == null
                 ? assembly.GetTypes()
                 : typesToCheck.Select(type => Type.GetType(type)).ToArray();
 
             return GetMefExportsForTypes(types);
         }
 
+        private Assembly LoadAssembly(string assemblyPath)
+        {
+            Assembly assembly = null;
+            var assemblyFilename = Path.GetFileNameWithoutExtension(assemblyPath);
+
+            try
+            {
+                assembly = Assembly.Load(assemblyFilename);
+                _logger.Trace($"'{assemblyFilename}' loaded from '{assembly.Location}'.");
+            }
+            catch (Exception e)
+            {
+                _logger.Trace($"'{assemblyFilename}' could not by loaded from probing paths. ({e.GetType().Name}: {e.Message})");
+            }
+
+            if (assembly == null)
+            {
+                try
+                {
+                    assembly = Assembly.LoadFrom(assemblyPath);
+                    _logger.Trace($"'{assemblyFilename}' loaded from '{assembly.Location}' via explicit LoadFrom.");
+                }
+                catch (Exception e)
+                {
+                    _logger.Trace($"'{assemblyFilename}' could not by loaded from '{assemblyPath}'. ({e.GetType().Name}: {e.Message})");

```

Comment:
Use the exception's ToString method, to include full info in the log:
`_logger.Trace($"'{assemblyFilename}' could not by loaded from '{assemblyPath}'. {e}");`